text/0000-multiarch-builders-and-package.md

```diff
@@ -0,0 +1,944 @@
+
+- Why should we do this?
+
+The uses of ARM architecture in the cloud and edge computing has been growing rapidly. The CNCF community has been also growing in the last years, and there is a need to support multi-arch for all the projects. The buildpacks community is not an exception, issues like:
+- [It would be nice to support easily creating a manifest list packages and builders](https://github.com/buildpacks/pack/issues/1460)
+- [Provide a way to specify desired platform when creating packages and builders](https://github.com/buildpacks/pack/issues/1459)
+- [Multi arch image build support](https://github.com/buildpacks/pack/issues/1570)
+
+Or the conversations around this topic in our [slack channel](https://cloud-native.slack.com/archives/C032LNSMY0P), even the [talk at Kubecon NA 2022](https://www.youtube.com/watch?v=Sdr5axlOnDI&list=PLj6h78yzYM2O5aNpRM71NQyx3WUe1xpTn&index=76) demonstrate the interest from the community in this feature.
+
+- What use cases does it support?
+
+Currently, buildpack authors can build and package their buildpacks for different OS and Architectures, but when they distribute them the URI for a buildpack can’t disambiguate, 
+they need to use different tags to differentiate between them. Tools like `docker buildx imagetools create` helps to create an [image index](https://github.com/opencontainers/image-spec/blob/master/image-index.md) to combine them but increasing their build pipelines complexity. 
+For example, take a look at this recent blog [post](https://deploy-preview-53--elegant-borg-5bd068.netlify.app/blog/steps-we-took-for-a-basic-arm64-support-in-buildpacks) or the [instructions](https://github.com/dmikusa/paketo-arm64/) created from @dmikusa to build an ARM64 builder.
+
+For those buildpack authors that are using `cross-compile` languages like [go](https://go.dev/) or [rust](https://www.rust-lang.org/) or maybe bash scripts, adding the capability to `pack buildpack package` and `pack builder create` to create multi-arch images 
+and also handle the creation of an [image index](https://github.com/opencontainers/image-spec/blob/master/image-index.md) will simplify their CI/CD pipelines and make the experience more suitable.
+
+- What is the expected outcome?
+
+`pack buildpack package` and `pack builder create` commands will be updated in a way that they will handle the creation of multi-arch OCI images
+
+# What it is
+[what-it-is]: #what-it-is
+
+The end-users for this proposal are **Buildpack authors**, we expect to improve their user experience when creating multi-architecture buildpacks and builders as follows
+
+## Multi-arch example
+
+Let's suppose a **Buildpack author** has a `buildpack.toml` created with a content similar to this one:
+
+```toml
+# Buildpack API version
+api = "0.12"
+
+# Buildpack ID and metadata
+[buildpack]
+  id = "examples/my-multiarch-buildpack"
+  version = "0.0.1"
+
+# List of targets operating systems, architectures and versions
+
+[[targets]]
```

Comment:
I am sorry for opening this can of worms, but technically "target" when used in the context of say `CNB_TARGET_OS` or `CNB_TARGET_ARCH` refers exclusively to the run image. Right now, all target data for the build image and the run image must match - but perhaps that won't always be the case? What will we do then?

Reply:
@natalieparellano what do you think we should use here?

Reply:
I am pretty sure that the current architecture will accommodate this. What would probably end up happening is that buildpack.toml would have a list of `[[targets]]` and the buildpack os/arch/bin directories would be a subset of that list. Then at build time (somehow, but probably in run.toml) you'd have a list of run images to fulfill all the os/arch combinations in `[[targets]]` and by iterating through that list, you have a multi-arch application image. The key is that the mapping between buildpack.toml `[[targets]]` and the buildpack os/arch/bin directories would not necessarily be 1:1, whereas today we assume it is.